src/Entity.php

```diff
@@ -730,4 +730,83 @@ public function delete() {
         $managerName::deleteEntity($this);
     }
 
+    /**
+     * @return array
+     */
+    public function metaData() {
+        $entityInformation = $this->entityInformation();
+        $propertiesInformation = $this->propertiesInformation();
+        $properties = [];
+
+        foreach ($propertiesInformation as $propertyInformation) {
+            $properties[$propertyInformation->name]["dataType"] = isset($propertyInformation->dataType) ? $propertyInformation->dataType->name() : null;
+            $properties[$propertyInformation->name]["persistable"] = $propertyInformation->persistable;
+
+            if ($relationship = $propertyInformation->relationship) {
+                if ($relationship instanceof \obo\Relationships\One) {
+                    $relationshipInformation = [
+                        "type" => "one",
+                        "entityClassNameToBeConnected" => $relationship->entityClassNameToBeConnected,
+                    ];
+
+                    if (\is_array($relationship->cascade) AND count($relationship->cascade)) {
+                        $relationshipInformation["cascade"] = \implode(", ", $relationship->cascade);
+                    }
+
+                    $relationshipInformation["autocreate"] = $relationship->autoCreate;
+
+                    if ($relationship->entityClassNameToBeConnectedInPropertyWithName) $relationshipInformation["entityClassNameToBeConnectedInPropertyWithName"] = $relationship->entityClassNameToBeConnectedInPropertyWithName;
+                    if ($relationship->connectViaProperty) $relationshipInformation["connectViaProperty"] = $relationship->connectViaProperty;
+                    if ($relationship->ownerNameInProperty) $relationshipInformation["ownerNameInProperty"] = $relationship->ownerNameInProperty;
+
+                } elseif ($relationship instanceof \obo\Relationships\Many) {
+                    $relationshipInformation = [
+                        "type" => "many",
+                        "entityClassNameToBeConnected" => $relationship->entityClassNameToBeConnected,
+                    ];
+
+                    if (\is_array($relationship->cascade)) {
+                        $relationshipInformation["cascade"] = \implode(", ", $relationship->cascade);
+                    }
+
+                    if ($relationship->connectViaPropertyWithName) $relationshipInformation["connectViaProperty"] = $relationship->connectViaPropertyWithName;
+                    if ($relationship->ownerNameInProperty) $relationshipInformation["ownerNameInProperty"] = $relationship->ownerNameInProperty;
+                    if ($relationship->connectViaRepositoryWithName) $relationshipInformation["connectViaRepositoryWithName"] = $relationship->connectViaRepositoryWithName;
+                    if ($relationship->sortVia) $relationshipInformation["sortVia"] = $relationship->sortVia;
+
```

Comment:
I would delete such empty lines here, but it is your call, I am not familiar with obo coding standard :D